migrations/9_create_user_emails_array.js

```diff
@@ -0,0 +1,49 @@
+const Settings = require('settings-sharelatex')
+const mongojs = require('mongojs')
+const db = mongojs(Settings.mongo.url, ['users'])
+const async = require('async')
+
+const handleExit = () => console.log('Got signal.  Shutting down.')
+process.on('SIGINT', handleExit)
+process.on('SIGHUP', handleExit)
+
+const initUserEmailsAttribute = (user, callback) => {
+  const update = {
+    $set: {
+      emails: [
+        {
+          email: user.email,
+          createdAt: new Date()
+        }
+      ]
+    }
+  }
+  db.users.update({ _id: user._id }, update, callback)
+}
+
+const updateAllUsersEmailsAttribute = (users, callback) => {
+  console.log(`updating ${user.length} users`)
+  async.eachSeries(users, initUserEmailsAttribute, callback)
+}
+
+exports.migrate = (client, done) =>
+  db.users.find(
+    { emails: { $exists: false } },
```

Comment:
I don't think there is an index on the existence of `emails`, but there is one on `emails.email`. You might find that it's faster to query on `{ 'emails.email': { $exists: false } }`, but I'm not 100% sure, You could try this on a Mongo secondary to verify?

Reply:
Is this optimisation needed? The scale of users in ServerPro is on the 10s to 100s

Reply:
Yeah, it's probably fine in that case.